extension/chrome/elements/passphrase.ts

```diff
@@ -147,7 +147,8 @@ View.run(class PassphraseView extends View {
     const pass = String($('#passphrase').val());
     const storageType: StorageType = ($('.forget-pass-phrase-checkbox').prop('checked') || this.orgRules.forbidStoringPassPhrase()) ? 'session' : 'local';
     let atLeastOneMatched = false;
-    for (const keyinfo of this.keysWeNeedPassPhraseFor!) { // if passphrase matches more keys, it will save the pass phrase for all keys
+    const allPrivateKeys = await KeyStore.get(this.acctEmail);
+    for (const keyinfo of allPrivateKeys) { // if passphrase matches more keys, it will save the pass phrase for all keys
       const prv = await KeyUtil.parse(keyinfo.private);
       try {
         if (await KeyUtil.decrypt(prv, pass) === true) {
```

Comment:
Looks good but may create side effects: `atLeastOneMatched` now means if it matched any keys at all, as opposed to any of the ones that are needed. I'd rename it to `matchedAtLeastOneKeyWeNeed`, and adjust the logic at `atLeastOneMatched = true;` appropriately to only set it to true if it has the same primary fingerprint as one of the keys in `this.keysWeNeedPassPhraseFor`.

Otherwise, you could have a situation when you have keys [A,B], you need key [A] but match [B]. The dialog will happily go away, yet the message still cannot be decrypted.

Reply:
Done, thank you 🙏 

Also added the test case, ready for review 🚀